index.bs

```diff
@@ -2874,13 +2888,47 @@ Dictionary {{AudioTimestamp}} Members</h5>
                 {{AudioRenderCapacity/update}} at {{AudioRenderCapacity}}, using
                 {{AudioRenderCapacityEvent}}, with the given update interval in
                 {{AudioRenderCapacityOptions}}.
+
+                <div algorithm="audiorendercapacity-start">
+                    When invoked, perform the following steps:
+
+                    1. Set {{[[update interval]]}} to the value of
+                        <code>options.{{AudioRenderCapacityOptions/updateInterval}}
+                        </code>.
+                    1. Set {{[[is running]]}} to <code>true</code>.
+                    1. <a href="https://html.spec.whatwg.org/multipage/media.html#queue-a-media-element-task">
+                        Queue a media element task</a> to run the metric 
+                        reporting loop.
+                </div>
             
             : <dfn>stop()</dfn>
             ::
                 Stops metric collection and analysis. It also stops dispatching
                 {{AudioRenderCapacity/update}} events.
+
+                <div algorithm="audiorendercapacity-stop">
+                    When invoked, perform the following steps:
+
+                    1. Set {{[[is running]]}} to <code>false<code>.
+                </div>
         </dl>
 
+        <h6 id="AudioRenderCapacity-metric-reporting-loop">
+        Metric Reporting Loop</h6>
+
+        This algorithm can be invoked by {{AudioRenderCapacity/start}} method
+        to report the performance metric repreatedly.
+
+        1. If {{[[is running]]}} is <code>false</code>, abort this algorithm.
+        2. Create |E|, which is a new instance of {{AudioRenderCapacityEvent}},
+            with computed performance metrics.
```

Comment:
We might need to have something that stores the load values on the context as a private slot (e.g. a list) and perform the average/peak/underrun calculation, resets the list, fire the event, etc. so it's clear what's going on.

Besides, do we want this to be clocked off of the rendering clock or the main thread?

I can see things going both ways:
- after a delay of `[[update interval]]` in the audio context clock domain, the rendering algorithm queues a task with all the data to the main thread, continuously, while it's running. No event is fired when not running (e.g. context suspended).
- after a delay of `[[update interval]]` in the system clock clock-domain, the main thread gathers the performance data, computes and fires the event. If the `AudioContext` is suspended, we can decide to fire or not fire the event.

I'd lean towards the second solution, _not_ firing the event when suspended. I think it's more useful: even if underrunning horribly, authors can know.

But what happens if the graph is so big that not a single rendering quantum was completed in the interval? That easy to simulate, and we should make it so that it reports 100% of load or something like that (some software report > 100%, that's also well-defined what it means).

Reply:
So you proposal is to expand "computed performance metrics" with more detailed steps? Right?

+1 on the second option.

For your last question, the API doesn't really specify how the renderer can recover from such problem. So we might see different behavior depending on the implementation. Some will drop subsequent system audio callbacks until the renderer becomes responsive while others will continue to process subsequent callbacks and accumulate more buffer underruns.

All that aside, 1.0 (100%) is practical. I am not sure what users can do with a number beyond 1.0.